### PR TITLE
Implement VesperRewarder contract

### PR DESCRIPTION
**Pull request by marcelomorgado**
This PR implements the `VesperRewarder` contract that creates a VSP pool to give rewards to the `MasterChefV2` (farming 2x) stakers.

The logic is pretty similar to the `MasterChefV2` logic itself (to give `SUSHI` rewards).

The contract has the `onSushiReward` function that is called whenever a user interacts with the farming pool (i.e. `deposit`, `withdraw`, `harvest`, `withdrawAndHarvest` and `emergencyWithdraw`).

The contract was based on [their example](https://github.com/sushiswap/sushiswap/blob/master/contracts/mocks/ComplexRewarder.sol) and the [`AMP` token code](https://etherscan.io/address/0x50cce4bB91eE7E0897dD7F1DE3010E436852DC71#code)

Reference: https://dev.sushi.com/sushiswap/contracts/masterchefv2/adding-double-incentives

### CODE REVIEW

**Review by patidarmanoj10 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1 +1,59 @@
...
+This repository contains the Vesper rewarder contract for Sushi 2x rewards.
+
+## Overview
+
+The `VesperRewarder` contract implements double incentive mechanism for SushiSwap farm feature.
+Whenever a user interacts with the `MasterChefV2` contract (that handle 2x rewards farming pools), the `VesperRewarder.onSushiReward()` function is called, and, that function calculates the VSP reward amount based on how much SLP token he staked and send it to the user.
+
+See more: https://dev.sushi.com/sushiswap/contracts/masterchefv2/adding-double-incentives
+
+## Install
```
Comment (jcvernaleo) [inline README.md:18]: Since you have a .nvmrc file and specify a version, I think readme should mention the `nvm use` step here.

**Review by jcvernaleo — CHANGES_REQUESTED** [review]
One minor extra thing I'd like to see in the README but otherwise looking good to me.

**Review by rokso — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (marcelomorgado) [reply README.md:18]: Done.

**Review by jcvernaleo — APPROVED** [review]
OK

**Pull request merged**